src/app/src/components/Quiz.js

```diff
@@ -41,34 +41,48 @@ class Quiz extends React.Component {
             currentResult: null,
             results: [],
             question: 0,
+            score: 0,
             answers,
             choices,
         };
     }
 
     checkResults = result => {
-        this.setState({
+        let pointsGained = 0;
+
+        const { numWrong, usedHint } = result;
+        if (numWrong === 0 && !usedHint) {
+            pointsGained += 100;
+        } else if (numWrong === 0) {
+            pointsGained += 80;
+        } else if (numWrong === 1) {
+            pointsGained += 60;
+        }
+
+        this.setState(prevState => ({
```

Comment:
Did not know that `setState` can be called with a function, with the previous state as a parameter. Cool.

Reply:
I learned that too 🤓 Makes sense to let React handle the state of `state` in case of simultaneous updates, https://itnext.io/react-setstate-usage-and-gotchas-ac10b4e03d60